### DISCUSSION TITLE
Split big component into multiple smaller components

### DISCUSSION THREAD

**Discussion by robert-stanciu** [livewire/livewire#5555] [1. Help] [answered]
Hi, community!

I have been using Livewire for over a year now and I feel I am doing something wrong when trying to split a bigger component into smaller ones.

This is what I got so far:

Big component concept:
- Has lots of properties, property listeners, validation rules, a huge HTML view, and so on. 
- The component's purpose is to collect user data from various inputs, show/hide different form sections based on form input, validate the inputs, and eventually store them in persistent storage.

Parent/child concept:
- Parent component responsibility would be to hide/show different form parts (Child components) based on form input, validate input data and eventually store input data.
- The child component (eventually, more than one) takes responsibility for validating its own properties, listening for property changes, and validating the input.

Issues I am facing with the Parent/child concept:
1. Two server requests when updating a child component. The first is the one that gets triggered because of wire:model on the Child component. The second one is because the Child component emits an event to its parent.
2. Duplicated validation is needed. One on the Parent component, just before storing the date. The second one (smaller because it contains only the child's properties) is on the child component and it makes sure the event is not emitted unless the validation passes.

Is there something I can improve or is there a completely different way to approach this concept?

**Comment by joshhanley** [comment, marked as answer]
@Dem0seQuence personally, I try to use as minimal Livewire components as possible, due to the issues you raised above. So I generally have one per page, say a `PostsListPage` component.

But I will sometimes use a nested component for things like the contents of a modal, like a `PostFormComponent` that gets shown in a modal on the posts list page.

This is because they are essentially unrelated, as such only need to fire an event from the form component to the list component, to tell the list to update as the data has changed.

For scenarios like you are talking about, where in reality it is just one big form with different sections, I would try to keep it all in the one component, so the interactivity all works as expected.

Hope this helps!

**Reply by robert-stanciu** [reply]
Hi,
thanks for the feedback. I personally do the modal thing the same way.

I ended up doing this big component and keeping everything in it.

Thanks!

**Reply by joshhanley** [reply]
No worries! 🙂
